src/mock_crypto.rs

```diff
@@ -107,14 +107,14 @@ pub(crate) mod crypto_impl {
             pub(crate) fn sign_detached(m: &[u8], sk: &SecretKey) -> Signature {
                 let mut temp = m.to_vec();
                 temp.extend(&sk.0);
-                Signature(hash64(&temp))
+                Signature(hashing_impl::sha3_256(&temp))
```

Comment:
To keep the speed up (and since this will need adjusted anyway if the signature length is increased to 64), we could maybe just put the result of `hash64` into the first 8 bytes and make the remainder all `0`, e.g.
```rust
let mut hash = hash64(&temp).to_vec();
hash.extend([0; 56].iter());
Signature(hash)
```